### DISCUSSION TITLE
Option<Option<T>> for optional nullable fields

### DISCUSSION THREAD

**Discussion by mlesin** [diesel-rs/diesel#3283] [Ideas]
I found very inconvenient to create a right changeset if I have a table with multiple nullable fields and I want to set only one of them to null (and do it with changeset to be able to use it in api).
I know there is a treat_none_as_null but it doesn't fit well in a such situation.
So my Idea is to use nested option type for such case, where first option will tell if this field is omitted or not, and the second one will tell if it has a value or should be null.
What do you think about it?
At first glance it seems like a breaking change, but looks like it worth it to make developers lives a bit better :)

**Comment by weiznich** [comment]
This is already what `#[derive(AsChangeset)]` does internally. Checkout the [All about updates](https://diesel.rs/guides/all-about-updates.html) guide for details.

**Reply by mlesin** [reply]
Oh my god, shame on me :)
Thanks a lot!
Maybe would be better to change the order of appearance of these two methods in docs? Because it seems initially I stopped reading  sentence after `#[changeset_options(treat_none_as_null="true")]`, don't know why :)

**Reply by weiznich** [reply]
We are always happy to receive PR's improving unclear documentation.
